src/binlog/events/transaction_payload_event.rs

```diff
@@ -0,0 +1,296 @@
+                                    format!("Error reading val for OTW_PAYLOAD_COMPRESSION_TYPE_FIELD"),
+                                ))?
+                                }
+                            };
+                            ob.algorithm =
+                                TransactionPayloadCompressionType::try_from(val).unwrap();
+                            have_compression_type = true;
+                            continue;
+                        }
+                        Ok(TransactionPayloadFields::OTW_PAYLOAD_UNCOMPRESSED_SIZE_FIELD) => {
+                            let _length = match buf.read_lenenc_int() {
+                                Ok(_) => {}
+                                Err(_) => {
+                                    let length = buf.eat_lenenc_int();
+                                    buf.skip(length as usize);
+                                    Err(io::Error::new(
+                                    io::ErrorKind::InvalidData,
+                                    format!("Error reading length for OTW_PAYLOAD_UNCOMPRESSED_SIZE_FIELD"),
+                                ))?
+                                }
+                            };
+                            let val = match buf.read_lenenc_int() {
+                                Ok(val) => val,
+                                Err(_) => {
+                                    let length = buf.eat_lenenc_int();
+                                    buf.skip(length as usize);
+                                    Err(io::Error::new(
+                                    io::ErrorKind::InvalidData,
+                                    format!("Error reading val for OTW_PAYLOAD_UNCOMPRESSED_SIZE_FIELD"),
+                                ))?
+                                }
+                            };
+                            ob.uncompressed_size = RawInt::new(val);
+                            continue;
+                        }
+                        Err(val) => {
+                            let length = buf.eat_lenenc_int();
+                            buf.skip(length as usize);
+                            Err(io::Error::new(
+                                io::ErrorKind::InvalidData,
+                                format!("Unknown field type: {:?}", val),
+                            ))?;
+                        }
+                    }
+                }
+                Err(val) => {
+                    let length = buf.eat_lenenc_int();
+                    buf.skip(length as usize);
+                    Err(io::Error::new(
+                        io::ErrorKind::InvalidData,
+                        format!("Unknown field type: {:?}", val),
+                    ))?;
+                }
+            };
+        }
+
+        Ok(ob)
+    }
+}
+
+impl MySerialize for TransactionPayloadEvent<'_> {
+    fn serialize(&self, buf: &mut Vec<u8>) {
+        self.payload_size.serialize(&mut *buf);
```

Comment:
Looks like it must follow TLV serialization (see [here](https://github.com/mysql/mysql-server/commit/1e5ae348c6f760b16a3972d84956358606a30041#diff-184e9a7d8a58f080974e475d4199fe5c6da5518c8a2811cc5df5988c8f9e9797R158-R185)). The END_MARK must also be here.

Reply:
Indeed. Done